File: ios/LaneShadow/Views/Molecules/DiscoveryEmptyOverlay.swift
import LaneShadowTheme
import SwiftUI

// MARK: - Discovery Empty Overlay Component

/**
 * Discovery empty overlay molecule component
 *
 * Glassmorphic empty state overlay for zero route results.
 * Displays context-aware message with suggestion to adjust filters.
 * Semi-transparent design (map visible behind).
 *
 * Following React Native component from react-native/components/discovery/discovery-empty-overlay.tsx
 *
 * ## Design Tokens Used
 * - Colors:
 *   - Background: `theme.colors.surface.default` with 0.8 opacity (80%)
 * - Layout:
 *   - Full-screen ZStack with ignoresSafeArea
 *   - Safe area top padding for notch/status bar
 *   - Content padding: 24pt horizontal
 *   - Centered content with VStack
 * - Reuses existing LSEmptyState component internally
 *
 * ## Parameters
 * - visible: Whether overlay should be shown (false returns EmptyView)
 * - message: Context-aware message (default: "No routes in this area")
 * - suggestion: Suggestion subtitle (default: "Try adjusting your filters or zooming out")
 * - ctaLabel: Optional CTA button label
 * - onCtaPress: Optional CTA button callback
 * - testID: Optional test identifier for testing
 */
public struct LSDiscoveryEmptyOverlay: View {
    @Environment(\.theme) private var theme

    private let visible: Bool
    private let message: String
    private let suggestion: String
    private let ctaLabel: String?
    private let onCtaPress: (() -> Void)?
    private let testID: String

    private let contentPadding: CGFloat = 24

    public init(
        visible: Bool,
        message: String = "No routes in this area",
        suggestion: String = "Try adjusting your filters or zooming out",
        ctaLabel: String? = nil,
        onCtaPress: (() -> Void)? = nil,
        testID: String = "discovery-empty-overlay"
    ) {
        self.visible = visible
        self.message = message
        self.suggestion = suggestion
        self.ctaLabel = ctaLabel
        self.onCtaPress = onCtaPress
        self.testID = testID
    }

    // MARK: - Body

    public var body: some View {
        if !visible {
            EmptyView()
        } else {
            GeometryReader { _ in
                ZStack {
                    // Semi-transparent background (80% opacity)
                    theme.colors.surface.default.opacity(0.8)
                        .ignoresSafeArea()

                    // Centered content with safe area top padding
                    VStack {
                        Spacer()

                        LSEmptyState(
                            icon: "map.marker.path",
                            headline: message,
                            body: suggestion,
                            ctaLabel: ctaLabel,
                            onCtaPress: onCtaPress
                        )
                        .padding(.horizontal, contentPadding)
                        .accessibilityIdentifier("\(testID)-state")

                        Spacer()
                    }
                    .accessibilityIdentifier(testID)
                }
            }
            .ignoresSafeArea()
        }
    }
}

// MARK: - Preview

#Preview("DiscoveryEmptyOverlay - Default") {
    ZStack {
        Color.gray.opacity(0.3) // Simulate map background

        LSDiscoveryEmptyOverlay(
            visible: true
        )
    }
    .laneShadowTheme()
}

#Preview("DiscoveryEmptyOverlay - Custom message") {
    ZStack {
        Color.gray.opacity(0.3) // Simulate map background

        LSDiscoveryEmptyOverlay(
            visible: true,
            message: "No routes match your filters",
            suggestion: "Try clearing your filters to see more results"
        )
    }
    .laneShadowTheme()
}

#Preview("DiscoveryEmptyOverlay - With CTA") {
    ZStack {
        Color.gray.opacity(0.3) // Simulate map background

        LSDiscoveryEmptyOverlay(
            visible: true,
            message: "No routes in this area",
            suggestion: "Try adjusting your filters or zooming out",
            ctaLabel: "Adjust Filters",
            onCtaPress: {
                print("CTA tapped")
            }
        )
    }
    .laneShadowTheme()
}

#Preview("DiscoveryEmptyOverlay - Hidden") {
    ZStack {
        Color.gray.opacity(0.3) // Simulate map background

        LSDiscoveryEmptyOverlay(
            visible: false
        )
    }
    .laneShadowTheme()
}

File: android/app/src/main/java/com/laneshadow/ui/components/molecules/DiscoveryEmptyOverlay.kt
package com.laneshadow.ui.components.molecules

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Search
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.semantics.contentDescription
import com.laneshadow.theme.LocalLaneShadowTheme

/**
 * DiscoveryEmptyOverlay molecule component
 *
 * Full-screen overlay with semi-transparent background for empty state
 * in the discovery screen. Wraps EmptyState with 80% opacity surface background.
 * Following React Native wrapper patterns from react-native/components/discovery/discovery-empty-overlay.tsx
 *
 * @param visible Whether the overlay is visible
 * @param message Optional headline message (default: "No routes in this area")
 * @param suggestion Optional body suggestion text (default: "Try adjusting your filters or zooming out")
 * @param ctaLabel Optional CTA button label
 * @param onCtaPress Optional CTA button callback
 * @param modifier Modifier for the component
 * @param testId Test ID for UI testing
 */
@Composable
fun DiscoveryEmptyOverlay(
    visible: Boolean,
    message: String = "No routes in this area",
    suggestion: String = "Try adjusting your filters or zooming out",
    ctaLabel: String? = null,
    onCtaPress: (() -> Unit)? = null,
    modifier: Modifier = Modifier,
    testId: String? = null,
) {
    val theme = LocalLaneShadowTheme.current

    // Build accessibility description
    val accessibilityDescription = if (visible) {
        "$message. $suggestion"
    } else {
        ""
    }

    // If not visible, render empty Box
    if (!visible) {
        Box(modifier = modifier)
        return
    }

    // Semi-transparent surface background (80% opacity = 0.8f alpha)
    val overlayBackgroundColor = theme.colors.surface.default.copy(alpha = 0.8f)

    Box(
        modifier = modifier
            .fillMaxSize()
            .background(overlayBackgroundColor)
            .testTag(testId ?: "discovery-empty-overlay")
            .semantics {
                contentDescription = accessibilityDescription
            },
        contentAlignment = Alignment.Center,
    ) {
        EmptyState(
            icon = Icons.Default.Search,
            headline = message,
            body = suggestion,
            ctaLabel = ctaLabel,
            onCtaClick = onCtaPress,
            testId = testId?.let { "$it-empty-state" } ?: "discovery-empty-overlay-empty-state",
        )
    }
}
